pkg/plugin/plugin_manager.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugin
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+
+	jvspb "github.com/abcxyz/jvs/apis/v0"
+	"github.com/abcxyz/pkg/multicloser"
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/go-plugin"
+)
+
+const (
+	// PluginGlob is the glob pattern used to find plugins.
+	PluginGlob = "jvs-plugin-*"
+)
+
+// PluginManager will discover and load plugins.
+type PluginManager struct {
+	// dir is the directory where plugins can be found.
+	dir string
+}
```

Comment:
It doesn't look like we need a struct?

```go
func LoadPlugins(dir string) (map[string]jvspb.Validator, *multicloser.Closer, error) {}
```

Reply:
Is there any way we can unit test this? (could be a follow up)

Reply:
I was thinking to store the validator map and multicloser to fields of the PluginManager, but it feels a bit awkward. Get rid of the struct for now.

Feels we will need to mock most of the plugin methods in order to test the logic. Probably better to handle it as a follow up. @sethvargo it would be great if you can point me to some existing unit test examples

Reply:
Resolving this thread now. I will have the unit tests in followed-up PR